### DISCUSSION TITLE
How can we implement a variable pressure generator?

### DISCUSSION THREAD

**Discussion by J-Anoop** [TS-CUBED/CirculatorySystemModels.jl#28] [Q&A]
Hello Dr. Torsten. I am trying to implement a LPM for coronary circulation using CSM.jl library. I watched your Juliacon2023 workshop on youtube, and was able to get up to speed. Appreciate your and Harry's efforts on that.  I am trying to reproduce results from an article  " Pietrabissa, Riccardo, et al. "A lumped parameter model to evaluate the fluid dynamics of different coronary bypasses." Medical engineering & physics 18.6 (1996): 477-484."  In there, they mention a variable pressure generator, which follows a similar pressure proile as a left ventricle. Its shown in an image below.
![image](https://github.com/TS-CUBED/CirculatorySystemModels.jl/assets/102414363/71834d07-61a0-41a5-9f36-4c4585d50920)
How do we implement this VPG? is it similar to for eg. a shichamber? but then wont it also contribute to flow there?

**Comment by TS-CUBED** [comment]
Hello. Interesting project.

If you have an external function that describes the pressure, you can use the `DrivenPressure` element for this.

Just define a function of t, and then pass that function as the `fun=yourFunction` argument.

Note that function needs to be differentiable, and you may need to @register it. I don't know the paper, so cannot say if the function is going to work.

Let me know how it goes and send me a small code snippet (MWE), then I can have a look.
